src/client.ts

```diff
@@ -3424,6 +3424,23 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         return this.getIgnoredUsers().includes(userId);
     }
 
+    /**
+     * Gets the list of currently ignored invites.
+     */
+    public getIgnoredInvites(): IgnoredInvites {
```

Comment:
also something I should have caught in the previous review was the usefulness of these functions: do we really need them, or can we just have the consumer call `get/setAccountData` on its own?

I realize that ignoring users is here already, but I honestly kinda question that too.

At the very least, I think the getter and setter should return a `{ [roomId]: Metadata }` object rather than the raw account data.

Reply:
> also something I should have caught in the previous review was the usefulness of these functions: do we really need them, or can we just have the consumer call get/setAccountData on its own?

In my mind, there are a few reasons to put it in the matrix-js-sdk:

- there may come a time when part of the spec is implemented by the server;
- the consumers for `getIgnoredInvites` and that for `setIgnoredInvites` look like they are going to be in very different parts of the code ([possibly in matrix-js-sdk](https://github.com/matrix-org/matrix-js-sdk/pull/2496)), so having a single source of truth with a single type signature, rather than inlining two untyped calls to `get/setAccountData` feels less accident-prone;

> At the very least, I think the getter and setter should return a { [roomId]: Metadata } object rather than the raw account data.

I don't understand that line. Do you mean that I should make a shallow copy of the account data rather than trusting the `readonly` in the type signature? A deep copy?

Reply:
should be the equivalent of `this.getAccountData("whatever")?.ignored_invites` - no explicit copying needed. The idea is to return the inner structure because the caller shouldn't be concerned with the intricacies of how account data is formed.